rskj-core/src/test/java/org/ethereum/rpc/exception/RskErrorResolverTest.java

```diff
@@ -0,0 +1,128 @@
+        Assert.assertNull(result.data);
+    }
+
+    @Test
+    public void test_resolveError_givenUnrecognizedPropertyException_nullExceptionMessage_returnsJsonErrorWithDefaultMessageAsExpected() throws NoSuchMethodException {
+        // Given
+        UnrecognizedPropertyException exception = mock(UnrecognizedPropertyException.class);
+        when(exception.getMessage()).thenReturn(null);
+
+        Method methodMock = this.getClass().getMethod("mockMethod");
+        List<JsonNode> jsonNodeListMock = new ArrayList<>();
+
+        // When
+        JsonError result = rskErrorResolver.resolveError(exception, methodMock, jsonNodeListMock);
+
+        // Then
+        Assert.assertNotNull(result);
+        Assert.assertEquals(-32603, result.code);
+        Assert.assertEquals("Invalid parameters", result.message);
+        Assert.assertNull(result.data);
+    }
+
+    @Test
+    public void test_resolveError_givenUnrecognizedPropertyException_returnsJsonErrorWithDescriptiveMessageAsExpected() throws NoSuchMethodException {
+        // Given
+        String message = "Unrecognized field \"form\" (class org.ethereum.rpc.Web3$CallArguments), not marked as ignorable (8 known properties: \"gasPrice\", \"value\", \"gas\", \"from\", \"to\", \"chainId\", \"nonce\", \"data\"])\n at [Source: N/A; line: -1, column: -1] (through reference chain: org.ethereum.rpc.Web3$CallArguments[\"form\"])";
+        UnrecognizedPropertyException exception = mock(UnrecognizedPropertyException.class);
+        when(exception.getMessage()).thenReturn(message);
+
+        Method methodMock = this.getClass().getMethod("mockMethod");
+        List<JsonNode> jsonNodeListMock = new ArrayList<>();
+
+        // When
+        JsonError result = rskErrorResolver.resolveError(exception, methodMock, jsonNodeListMock);
+
+        // Then
+        Assert.assertNotNull(result);
+        Assert.assertEquals(-32603, result.code);
+        Assert.assertEquals("Unrecognized field \"form\" (class org.ethereum.rpc.Web3$CallArguments), not marked as ignorable (8 known properties: \"gasPrice\", \"value\", \"gas\", \"from\", \"to\", \"chainId\", \"nonce\", \"data\"])", result.message);
+        Assert.assertNull(result.data);
+    }
+
+    @Test
+    public void test_resolveError_givenGenericException_returnsJsonErrorWithDefaultMessageAsExpected() throws NoSuchMethodException {
+        // Given
+        Exception exception = mock(Exception.class);
+
+        Method methodMock = this.getClass().getMethod("mockMethod");
+        List<JsonNode> jsonNodeListMock = new ArrayList<>();
+
+        // When
+        JsonError result = rskErrorResolver.resolveError(exception, methodMock, jsonNodeListMock);
+
+        // Then
+        Assert.assertNotNull(result);
+        Assert.assertEquals(-32603, result.code);
+        Assert.assertEquals("Internal server error", result.message);
+        Assert.assertNull(result.data);
+    }
+
+    public void mockMethod() { }
```

Comment:
Maybe I'd make this private (unless it's necessary because of reflection magics)

Reply:
In this case it is necessary so it is easier to use, will try a couple of other options and let you know.